src/wp-includes/theme.php

```diff
@@ -1540,9 +1547,33 @@ function get_uploaded_header_images() {
 		}
 	}
 
+	set_transient( $transient_key, $header_images, DAY_IN_SECONDS );
+
 	return $header_images;
 }
 
+/**
+ * Invalidates the cache for header images when a custom header image is changed or deleted.
+ *
+ * @since 6.7
+ *
+ * @param int $meta_id The meta ID of the attachment meta data.
+ * @param int $post_id The post ID of the attachment.
+ * @param string $meta_key The meta key of the attachment meta data.
+ * @param mixed $meta_value The new value of the attachment meta data.
+ *
+ * @return void
+ */
+function invalidate_header_images_cache( $meta_id, $post_id, $meta_key, $meta_value ) {
+	if ( '_wp_attachment_is_custom_header' === $meta_key ) {
+		$stylesheet    = get_option( 'stylesheet' );
```

Comment:
This needs to respect `$meta_value`. In most cases, it'll be the same as the current stylesheet, but it could also be updated for another theme than the current theme - unlikely, but possible. At the moment you aren't using `$meta_value` at all.

Note that we also need to still cater for the case that `$meta_value` is empty (e.g. upon deletion), so we should use the current stylesheet only as a fallback for that case.
```suggestion
		$stylesheet    = $meta_value ? $meta_value : get_option( 'stylesheet' );
```